Makefile

```diff
@@ -1,23 +1,27 @@
 test:
-	go test -timeout 20s -race -coverprofile coverage.txt -covermode=atomic ./...
```

Comment:
would be good to keep -race this will check data race conditions, especially since you are using channels

Reply:
it's adding 7 or 8 seconds to each run, should I separate this out into some other make instruction?

Reply:
i guess you could, then setup travis to run the race detector

Reply:
added it back in.  I can live with 9 seconds. :)